src/services/edusharing/logic/connector.js

```diff
@@ -0,0 +1,243 @@
+		return !!this.authorization && !!this.accessToken;
+	}
+
+	async GET(id, params) {
+		if (!this.allConfigurationValuesHaveBeenDefined()) {
+			throw this.configurationIncompleteError();
+		}
+
+		if (this.isLoggedin() === false) {
+			await this.login();
+		}
+
+		const propertyFilter = '-all-';
+
+		const options = {
+			method: 'GET',
+			// eslint-disable-next-line max-len
+			url: `${Configuration.get('ES_DOMAIN')
+			}/edu-sharing/rest/node/v1/nodes/mv-repo.schul-cloud.org/${id
+			}/metadata?propertyFilter=${propertyFilter}`,
+			headers: {
+				...EduSharingConnector.headers,
+				cookie: this.authorization,
+			},
+			timeout: REQUEST_TIMEOUT,
+		};
+
+		const eduResponse = await this.requestRepeater(options);
+		return eduResponse.node;
+	}
+
+	async FIND(data) {
+		const contentType = data.query.contentType || 'FILES'; // enum:[FILES,FILES_AND_FOLDERS,COLLECTIONS,ALL]
+		const skipCount = parseInt(data.query.$skip, 10) || 0;
+		const maxItems = parseInt(data.query.$limit, 10) || 9;
+		const sortProperties = data.query.sortProperties || 'score';
+		const sortAscending = false;
+		const propertyFilter = '-all-'; // '-all-' for all properties OR ccm-stuff
+		const searchWord = data.query.searchQuery || '';
+
+		if (!this.allConfigurationValuesHaveBeenDefined()) {
+			throw this.configurationIncompleteError();
+		}
+
+		if (this.isLoggedin() === false) {
+			await this.login();
+		}
+
+		const urlBase = `${Configuration.get('ES_DOMAIN')
+		}/edu-sharing/rest/search/v1/queriesV2/mv-repo.schul-cloud.org/mds/ngsearch/?`;
+		const url = urlBase
+			+ [
+				`contentType=${contentType}`,
+				`skipCount=${skipCount}`,
+				`maxItems=${maxItems}`,
+				`sortProperties=${sortProperties}`,
+				`sortAscending=${sortAscending}`,
+				`propertyFilter=${propertyFilter}`,
+			].join('&');
+
+		const options = {
+			method: 'POST',
+			// This will be changed later with a qs where sorting, filtering etc is present.
+			// eslint-disable-next-line max-len
+			url,
+			headers: {
+				...EduSharingConnector.headers,
+				cookie: this.authorization,
+			},
+			body: JSON.stringify({
+				criterias: [
+					{ property: 'ngsearchword', values: [`${searchWord}`] },
+				],
+				facettes: ['cclom:general_keyword'],
+			}),
+			timeout: REQUEST_TIMEOUT,
+		};
+
+		const parsed = await this.requestRepeater(options);
+
+		// adds accesstoken to image-url to let user see the picture on client-side.
+		if (parsed && parsed.nodes) {
+			parsed.nodes.forEach((node) => {
+				if (node.preview && node.preview.url) {
+					node.preview.url += `&accessToken=${this.accessToken}`;
+				}
+			});
+		}
+
+		return {
+			total: parsed.pagination.total,
+			limit: parsed.pagination.count,
+			skip: parsed.pagination.from,
+			data: parsed.nodes,
+		};
+	}
+
+
+	static get Instance() {
+		if (!EduSharingConnector.instance) {
+			return new EduSharingConnector();
+		}
+		return EduSharingConnector.instance;
+	}
+}
+
+module.exports = EduSharingConnector.Instance;
```

Comment:
doesn't doing that the same like the explicit Singleton?
```suggestion
module.exports = new EduSharingConnector();
```

Reply:
ok